x-pack/platform/plugins/shared/ai_infra/product_doc_base/server/services/package_installer/utils/semver.ts

```diff
@@ -7,15 +7,23 @@
 
 import Semver from 'semver';
 
-export const latestVersion = (versions: string[]): string => {
-  let latest: string = versions[0];
-  for (let i = 1; i < versions.length; i++) {
-    const current = versions[i];
-    if (Semver.gt(Semver.coerce(current)!, Semver.coerce(latest)!)) {
-      latest = current;
+export const latestVersion = (availableVersions: string[], kibanaVer?: string): string => {
+  const kibanaSemver = kibanaVer ? Semver.coerce(kibanaVer) : undefined;
+  let latest: string | undefined;
+
+  for (const version of availableVersions) {
+    const semver = Semver.coerce(version);
+    if (!semver) continue;
+
+    if (
+      (!kibanaSemver || Semver.lte(semver, kibanaSemver)) &&
```

Comment:
Nit: Comment might be helpful here to clarify what is being checked.

Reply:
Added comment here for clarity [`6db7dac` (#229704)](https://github.com/elastic/kibana/pull/229704/commits/6db7dacd2bae6aa129adfd1f697a7c7b504c5f2e)